src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js

```diff
@@ -0,0 +1,182 @@
+                <FormattedMessage
+                  description="Variable prefill attribute label"
+                  defaultMessage="Attribute"
+                />
+              }
+              errors={errors.attribute}
+            >
+              <AttributeField />
+            </Field>
+          </FormRow>
+
+          <FormRow>
+            <Field
+              name="identifierRole"
+              label={
+                <FormattedMessage
+                  description="Variable prefill identifier role label"
+                  defaultMessage="Identifier role"
+                />
+              }
+              errors={errors.identifierRole}
+            >
+              <IdentifierRoleField />
+            </Field>
+          </FormRow>
+        </Fieldset>
+
+        <SubmitRow>
+          <SubmitAction
+            onClick={event => {
+              event.preventDefault();
+              handleSubmit(event);
+            }}
+          />
+        </SubmitRow>
+      </>
+    )}
+  </Formik>
+);
+
+PrefillConfigurationForm.propTypes = {
+  onSubmit: PropTypes.func.isRequired,
+  plugin: PropTypes.string,
+  attribute: PropTypes.string,
+  identifierRole: PropTypes.string,
+  errors: PropTypes.shape({
+    plugin: PropTypes.arrayOf(PropTypes.string),
+    attribute: PropTypes.arrayOf(PropTypes.string),
+    identifierRole: PropTypes.arrayOf(PropTypes.string),
+  }).isRequired,
+};
+
+const PluginField = () => {
+  const [fieldProps] = useField('plugin');
+  const {setFieldValue} = useFormikContext();
+  const {
+    plugins: {availablePrefillPlugins},
+  } = useContext(FormContext);
+
+  const {value} = fieldProps;
+
+  // reset the attribute value whenever the plugin changes
+  useUpdateEffect(() => {
+    setFieldValue('attribute', '');
+  }, [setFieldValue, value]);
+
+  const choices = availablePrefillPlugins.map(plugin => [plugin.id, plugin.label]);
+  return <Select allowBlank choices={choices} id="id_plugin" {...fieldProps} />;
+};
+
+const AttributeField = () => {
+  const [fieldProps] = useField('attribute');
+  const {
+    values: {plugin},
+  } = useFormikContext();
+
+  // Load the possible prefill attributes
+  // XXX: this would benefit from client-side caching
+  const {
+    loading,
+    value = [],
+    error,
+  } = useAsync(async () => {
+    if (!plugin) return [];
+
+    const endpoint = `/api/v2/prefill/plugins/${plugin}/attributes`;
+    // XXX: clean up error handling here at some point...
+    const response = await get(endpoint);
+    if (!response.ok) throw response.data;
+    return response.data.map(attribute => [attribute.id, attribute.label]);
+  }, [plugin]);
+
+  // throw errors to the nearest error boundary
+  if (error) throw error;
+  const choices = loading ? LOADING_OPTION : value;
+  return (
+    <Select
+      allowBlank
+      choices={choices}
+      id="id_attribute"
+      disabled={loading || !plugin}
+      {...fieldProps}
+    />
+  );
+};
+
+const IdentifierRoleField = () => {
```

Comment:
the problem is that there is actually no value set in the state at all, and that's why the `allowBlank` is needed. This role can indeed be empty, especially with legacy form exports created before this field was added. The backend code also falls back to `main`, but I'll take a look if I can improve this.

Reply:
the validation error styling was being mangled by bootstrap, again... So patching this up, and in this case the `allowBlank` can actually be removed since we normalize the formik `initialValues` to address potential missing information.